### ISSUE TITLE
[C#] Implement new Concept Exercise: booleans

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/v3#1004]
This issue describes how to implement the `booleans` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the Concept of Booleans in C#.

## Learning objectives

- Know of the existence of the `bool` type and its two values.
- Know about boolean operators and how to build logical expressions with them.
- Know of the boolean operator precedence rules.

## Out of scope

Nothing.

## Concepts

This Concepts Exercise's Concepts are:

- `booleans`: know of the existence of the `bool` type and its two values; know about boolean operators and how to build logical expressions with them; know of the boolean operator precedence rules.

## Prequisites

This exercise's prerequisites Concepts are:

- `basics`: know how to define methods.

### Hints

- [Booleans][booleans]: introduces the `bool` type.
- [Boolean operators][boolean-operators]: describes the boolean operators.
- [Boolean operator precedence][boolean-operator-precedence]: lists the boolean operator precedence rules.

### After

- [Booleans][booleans]: introduces the `bool` type.
- [Boolean operators][boolean-operators]: describes the boolean operators.
- [Boolean operator precedence][boolean-operator-precedence]: lists the boolean operator precedence rules.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-doc]: https://github.com/exercism/v3/blob/master/reference/types/boolean.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[booleans]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/bool
[boolean-operators]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators
[boolean-operator-precedence]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#operator-precedence

**Comment by batibot323** [comment]
I'm interested in doing this after reading the [F# implementation](https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/booleans). Can I assign to myself?

**Comment by ErikSchierboom** [comment]
Sure!

**Comment by batibot323** [comment]
I'm sorry guys. I'll unassign myself from this first. I can't find the time to do this. I hope to come back soon! :)

**Comment by ErikSchierboom** [comment]
No worries. I can pick this up. Thanks for letting us know.
